### PR TITLE
Makeshift Mallet

### PR DESCRIPTION
**Pull request by ghost**
"That's it, i'm gettin me mallet!"

slapcrafted much like spears, but with an empty toolbox. it does an OK amount of damage, 14, less than a spear but it's wielded damage is massive. The thing is, one hit wielded will also break the mallet foreverrrr! (or, at least until you make another one)

:cl: Armhulen and Onule, dream team
add: You can now make mallets much like how you would make a spear!
add: Wield it for great power, but at what cost?
/:cl:

### CODE REVIEW

**Comment by DaxDupont** [comment]
30 force is really high though, even past fireaxe levels

**Comment by Armhulen** [comment]
it's one use, and you'd be getting more from two spear hits.

**Comment by DaxDupont** [comment]
I misread that part, disregard.

File: code/game/objects/items/twohanded.dm [hunk 1]
```diff
@@ -485,6 +486,43 @@
...
 	update_icon()
 
+//maul
+/obj/item/twohanded/maul
+	icon_state = "toolboxmaulr0"
+	lefthand_file = 'icons/mob/inhands/weapons/hammers_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/hammers_righthand.dmi'
+	name = "makeshift maul"
+	desc = "A haphazardly-constructed yet still deadly weapon of ancient design."
+	force = 10
```
Comment (KorPhaeron) [inline code/game/objects/items/twohanded.dm]: Why is this different than force unwielded?

Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: i fixed that in my next commit sorry

**Comment by Armhulen** [comment]
i really might remove this from the quick crafting menu and just make it slapcrafting as it does this so much easier

**Comment by Iatots** [comment]
If you use a syndie toolbox, you should be able to fill it with grenades that prime and spread on impact.

**Comment by praisenarsie** [comment]
@Armhulen sorry that's not allowed because haha our current crafting system is TOTALLY better in every single way!

**Comment by FantasticFwoosh** [comment]
- Itt someone pray/admin abuses a var-edited unbreakable 1 shot mallet for valids

**Comment by Armhulen** [comment]
it's hardcoded to not break but i can make a mtoolnir that doesn't ;)

**Comment by Armhulen** [comment]
also how is 30 damage 1 shot

**Comment by praisenarsie** [comment]
this is way worse than a spear

**Comment by PKPenguin321** [comment]
base damage is a bit much, kind of invalidates the spear

**Comment by Armhulen** [comment]
13 vs spear's 14 > 18

you'd be doing less unless you want to completely break your weapon

**Comment by PKPenguin321** [comment]
eh youre probably right

**Comment by Fox-McCloud** [comment]
Even if it it breaks after one hit, a 40% chance to knock someone down shouldn't be taken lightly; 30 damage, followed up by a welder or hatchet is likely going to keep someone permastunned.

**Comment by jughu** [comment]
A IED on a spear is probaly gonna do the same fox

**Comment by praisenarsie** [comment]
@jughu is correct

again, i'm gonna say again, this is way worse than a spear or ied.

**Comment by Fox-McCloud** [comment]
Yeah, it will.

I don't think we should use that as a balance talking point though; grenade spears are prettttty terrible.

**Comment by jughu** [comment]
eh its all chancebased and will come the same point

File: code/game/objects/items/twohanded.dm [hunk 2]
```diff
@@ -485,6 +486,84 @@
...
+	righthand_file = 'icons/mob/inhands/weapons/hammers_righthand.dmi'
+	name = "makeshift mallet"
+	desc = "A haphazardly-constructed yet still deadly weapon of ancient design. Wield for a critical hit, to your opponent and your mallet."
+	force = 14
+	w_class = WEIGHT_CLASS_HUGE
+	force_unwielded = 14
+	force_wielded = 30
+	throwforce = 15
+	throw_speed = 2
+	materials = list(MAT_METAL=2000)
```
Comment (swindly) [inline code/game/objects/items/twohanded.dm]: This is more than the sum of its parts.

Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: you'd think toolboxes were worth more

**Comment by Armhulen** [comment]
at the end of the day they both have upsides and downsides on what you can do with them. you'll probably be killing faster with the spear but 30 damage burst is very nice to have in some cases.

**Comment by Armhulen** [comment]
the fact that people can't decide if this is balanced to being underpowered or overpowered brings me a sense of pride and satisfaction

**Comment by Armhulen** [comment]
@Fox-McCloud post

**Comment by Armhulen** [comment]
i might be forgetting something but this is ready finally

**Comment by DoctoreMad** [comment]
You can make unbalanced shit that people can't figure out if its underpowered or overpowered easily. Just intentionally try to make something overpowered and then balance it out by giving it some huge penalty

It legitimately seems to never work out well.

**Comment by Armhulen** [comment]
ok so why is it unbalanced @DoctoreMad

**Comment by optimumtact** [comment]
great another stun wow so cool man you make so many original and cool weapons and idea wow so great

**Comment by DoctoreMad** [comment]
I can't wait to never, ever be able to find a toolbox again because the assistants are all playing whackamole

**Comment by DoctoreMad** [comment]
@Armhulen Because you're trying to balance a public 30 damage stun by giving it a crafting time?

**Comment by imsxz** [comment]
I don't see how people could see this as overpowered. The spear is gonna end up doing more damage over time due to the fact that the mallet breaks after one wielded hit. It's not even a guaranteed stun either unless I misread something. Additionally, you can't carry these on your back the same way you can with spears.

**Comment by Armhulen** [comment]
14 > 18 damage weapon you can make with a glass shard and a rod, can use forever can attach grenades can impale

vs

13 > 30 damage weapon that you have to carry like a chainsaw (no back storage) and it breaks after one use, 40% to stun on the heavy hit

**Comment by Armhulen** [comment]
i think the point is that they have different upsides and downsides and some will prefer one over the other!

**Comment by Bawhoppen** [comment]
it's just kind of weird

File: code/game/objects/items/weaponry.dm [hunk 3]
```diff
@@ -253,6 +258,65 @@
 		qdel(src)
 
 		user.put_in_hands(P)
+//MALLETS BELOW//
+	else if(istype(I, /obj/item/storage/toolbox/mechanical) && !(I.flags_1 & NODROP_1))
```
Comment (RemieRichards) [inline code/game/objects/items/weaponry.dm]: This code is making me physically ill.

File: code/game/objects/items/twohanded.dm [hunk 4]
```diff
@@ -485,6 +486,89 @@
...
+	..()
+	if(wielded)
+		integ--
+	if(!integ)
+		user.visible_message("<span class='notice'>[src] smashes into pieces!</span>")
+		qdel(src)
+		var/obj/item/wirerod/W = new /obj/item/wirerod
+		user.put_in_hands(W)
+
+/obj/item/twohanded/mallet/b
```
Comment (AnturK) [inline code/game/objects/items/twohanded.dm]: Stop saving on characters.

File: code/game/objects/items/twohanded.dm [hunk 5]
```diff
@@ -485,6 +486,89 @@
...
+	hitsound = 'sound/weapons/smash.ogg'
+	attack_verb = list("smacked", "cracked", "attacked")
+	max_integrity = 50
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 50, acid = 30)
+	var/integ = 1 //different toolboxes get different amount of hits before breaking, only used for the syndicate mallet and mtoolnir
+
+/obj/item/twohanded/mallet/update_icon()
+	icon_state = "toolbox_mallet_r[wielded]"
+
+/obj/item/twohanded/mallet/attack(atom/movable/AM, mob/user, proximity)
```
Comment (AnturK) [inline code/game/objects/items/twohanded.dm]: after attack is literally made for this.

File: code/game/objects/items/weaponry.dm (hunk 3 above)
Comment (AnturK) [reply code/game/objects/items/weaponry.dm]: I feel you.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by Armhulen** [comment]
AnturK 17 hours ago  •  edited  Member
I feel you.

For some constructive advice , reduce copypasta, use proper helpers for removing from hands, try to kill the if tree

ok i'll get right to wo-

**Pull request closed without merging**